dsub.py

```diff
@@ -627,237 +351,78 @@ def get_job_metadata(args, script, provider):
-  but with associated values coming from the data rows.
-
-  Environment variables columns are headered as "--env <name>"
-  Inputs columns are headered as "--input <name>" with the name optional.
-  Outputs columns are headered as "--output <name>" with the name optional.
-
-  For historical reasons, bareword column headers (such as "JOB_ID") are
-  equivalent to "--env var_name".
-
-  Args:
-    header: The first row of the tab-delimited jobs table file.
-    input_file_param_util: Utility for producing InputFileParam objects.
-    output_file_param_util: Utility for producing OutputFileParam objects.
-
-  Returns:
-    job_params: A list of EnvParams and FileParams for the environment
-    variables, input file parameters, and output file parameters.
-
-  Raises:
-    ValueError: If a header contains a ":" and the prefix is not supported.
-  """
-  job_params = []
-
-  # Tokenize the header line and process each field
-  header_columns = header.split('\t')
-  for col in header_columns:
-
-    # Reserve the "-" and "--" namespace.
-    # If the column has no leading "-", treat it as an environment variable
-    col_type = '--env'
-    col_value = col
-    if col.startswith('-'):
-      col_type, col_value = split_pair(col, ' ', 1)
-
-    if col_type == '--env':
-      job_params.append(EnvParam(col_value))
-
-    elif col_type == '--input' or col_type == '--input-recursive':
-      name = input_file_param_util.get_variable_name(col_value)
-      job_params.append(
-          InputFileParam(
-              name, recursive=(col_type.endswith('recursive'))))
-
-    elif col_type == '--output' or col_type == '--output-recursive':
-      name = output_file_param_util.get_variable_name(col_value)
-      job_params.append(
-          OutputFileParam(
-              name, recursive=(col_type.endswith('recursive'))))
-
-    else:
-      raise ValueError('Unrecognized column header: %s' % col)
+def tasks_to_job_ids(provider, task_list):
+  """Returns the set of job IDs for the given tasks."""
+  return set([provider.get_job_field(t, 'job-id') for t in task_list])
 
-  return job_params
 
+def wait_after(provider, jobid_list, poll_interval, stop_on_failure):
+  """Print status info as we wait for those jobs.
 
-def table_to_job_data(path, input_file_param_util, output_file_param_util):
-  """Parses a table of parameters from a TSV.
+  Blocks until either all of the listed jobs succeed,
+  or one of them fails.
 
   Args:
-    path: Path to a TSV file with the first line specifying the environment
-    variables, input, and output parameters as column headings. Subsequent
-    lines specify parameter values, one row per job.
-    input_file_param_util: Utility for producing InputFileParam objects.
-    output_file_param_util: Utility for producing OutputFileParam objects.
+    provider: job service provider
+    jobid_list: a list of job IDs (string) to wait for
+    poll_interval: integer seconds to wait between iterations
```

Comment:
The program would crash if given fractional seconds, so I'd suggest keeping the word "integer".